### PR TITLE
Remove noisy logs while syncing the blocks

### LINKED ISSUE #92: Remove noisy logs while syncing the blocks
Syncing the blocks produces a lot of unnecessary noise

### Expected behavior

- When there are no blocks to sync, a log is produced stating that it is synced.
- When a reorganization occurs, a log should be produced which includes the cut point hash (the least common block in the two chains) as well as the latest block in the new chain.
- When syncing several blocks, a log must be produced each time we are in a block height that is multiple of 5000, also, a log should be produced stating the the sync is done.

### Actual behavior

- The initial block syncing produces more than a million logs!
- The process usually looks for new blocks every minute, 

### Steps to reproduce the behavior
- Start the service having an empty database.

Please list all relevant steps to reproduce the observed behavior.

### PR DESCRIPTION
**Pull request by iBlackShadow**
Fixes #92

### CODE REVIEW

**Comment by adinael** [comment]
The commit message must have the prefix **server:**  See[ CONTRIBUTING.md ](https://github.com/X9Developers/block-explorer/blob/develop/CONTRIBUTING.md#commits)

File: server/test/com/xsn/explorer/services/LedgerSynchronizerServiceSpec.scala [hunk 1]
```diff
@@ -101,7 +101,8 @@ class LedgerSynchronizerServiceSpec extends PostgresDataHandlerSpec with BeforeA
         genesis,
```
Comment (adinael) [inline server/test/com/xsn/explorer/services/LedgerSynchronizerServiceSpec.scala]: Unnecessary changes

File: server/app/com/xsn/explorer/services/LedgerSynchronizerService.scala [hunk 2]
```diff
@@ -207,9 +224,9 @@ class LedgerSynchronizerService @Inject()(
     } yield {
       if (transactions.size != filteredTransactions.size) {
         // see https://github.com/bitpay/insight-api/issues/42
-        logger.warn(
-          s"The block = ${rpcBlock.hash} has phantom ${transactions.size - filteredTransactions.size} transactions, they are being discarded"
-        )
+        // logger.warn(
```
Comment (adinael) [inline server/app/com/xsn/explorer/services/LedgerSynchronizerService.scala]: remove commented code

File: server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    synchedBlocksCount: Int,
+    reorganization: Boolean,
+    reorganizationCuttingPoint: Option[Block],
+    reorganizationCuttingPointReached: Boolean
+) {
+
+  def syncCompleted(lastBlock: rpc.Block): LedgerSynchronizerServiceLogger = {
+    if (synchedBlocksCount == 0) {
+      logger.info(s"Ledger up to date at height ${lastBlock.height.int}")
+    } else {
```
Comment (AlexITC) [inline server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala:18]: `else if` avoids unnecessary nesting.

File: server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    this
+  }
+
+  def blockSynched(previousBlock: Option[Block], block: rpc.Block): LedgerSynchronizerServiceLogger = {
+    if (block.height.int % 5000 == 0) {
+      logger.info(s"Synced block ${block.hash} at height ${block.height.int}")
+    }
+
+    if (reorganization && !reorganizationCuttingPointReached) {
+      return this.copy(
```
Comment (AlexITC) [inline server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala:41]: avoid `return` statements.

File: server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      } else {
+        logger.info(s"Sync completed at height ${lastBlock.height.int}, $synchedBlocksCount blocks synched")
+      }
+    }
+
+    this
+  }
+
+  def blockSynched(previousBlock: Option[Block], block: rpc.Block): LedgerSynchronizerServiceLogger = {
+    if (block.height.int % 5000 == 0) {
```
Comment (AlexITC) [inline server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala:36]: Avoid magic numbers, use a constant instead.

File: server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    logger: Logger,
+    synchedBlocksCount: Int,
+    reorganization: Boolean,
+    reorganizationCuttingPoint: Option[Block],
+    reorganizationCuttingPointReached: Boolean
+) {
+
+  def syncCompleted(lastBlock: rpc.Block): LedgerSynchronizerServiceLogger = {
+    if (synchedBlocksCount == 0) {
+      logger.info(s"Ledger up to date at height ${lastBlock.height.int}")
```
Comment (AlexITC) [inline server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala:17]: a case class is not supposed to perform side effects, they are good for holding and transforming data.

File: server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        reorganizationCuttingPoint = previousBlock,
+        reorganizationCuttingPointReached = true,
+        synchedBlocksCount = synchedBlocksCount + 1
+      )
+    }
+    this.copy(synchedBlocksCount = synchedBlocksCount + 1)
+  }
+
+  def reorganize: LedgerSynchronizerServiceLogger = {
+    this.copy(reorganization = true)
```
Comment (AlexITC) [inline server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala:51]: +1

File: server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+  def trimed: LedgerSynchronizerServiceLogger = {
+    this.copy(reorganization = true)
+  }
+}
+
+object LedgerSynchronizerServiceLogger {
+
+  def apply(logger: Logger): LedgerSynchronizerServiceLogger = {
+    new LedgerSynchronizerServiceLogger(logger, 0, false, None, false)
```
Comment (AlexITC) [inline server/app/com/xsn/explorer/loggers/LedgerSynchronizerServiceLogger.scala:62]: Named parameters are a good fit, they would avoid possible confusion on the argument order.

File: server/app/com/xsn/explorer/services/LedgerSynchronizerService.scala [hunk 9]
```diff
@@ -205,12 +222,6 @@ class LedgerSynchronizerService @Inject()(
       validContracts <- getValidContracts(contracts).toFutureOr
       filteredTransactions = transactions.filter(_.blockhash == rpcBlock.hash)
     } yield {
-      if (transactions.size != filteredTransactions.size) {
```
Comment (AlexITC) [inline server/app/com/xsn/explorer/services/LedgerSynchronizerService.scala:208]: This log is important.

**Comment by AlexITC** [comment]
There is a simpler synchronization method that was introduced at 4e82f19c23523d4a4df130ef421621df74579f8b, it simplifies a lot the logging what's syncing, so, there is no need to merge this PR.

Thanks!
